rotkehlchen/exchanges/coinbase.py

```diff
@@ -1038,22 +1047,70 @@ def _deserialize_history_event(self, raw_data: dict[str, Any]) -> HistoryEvent |
                 )
 
             amount_data = raw_data.get('amount', {})
-            amount = deserialize_asset_amount(amount_data['amount'])
+            amount = deserialize_asset_amount_force_positive(amount_data['amount'])
             asset = asset_from_coinbase(amount_data['currency'], time=timestamp)
             notes = raw_data.get('details', {}).get('header', '')
             tx_type = raw_data['type']
+            event_identifier = f'{CB_EVENTS_PREFIX}{raw_data["id"]!s}'
+            timestamp_ms = ts_sec_to_ms(timestamp)
             if notes != '':
                 notes += f' {"from coinbase earn" if tx_type == "send" else "as " + tx_type}'
+
+            if tx_type in ('staking_transfer', 'unstaking_transfer'):
+                event_subtype = HistoryEventSubType.DEPOSIT_ASSET if tx_type == 'staking_transfer' else HistoryEventSubType.REMOVE_ASSET  # noqa: E501
+                return HistoryEvent(
+                    event_identifier=event_identifier,
+                    sequence_index=0,
+                    timestamp=timestamp_ms,
+                    location=Location.COINBASE,
+                    event_type=HistoryEventType.STAKING,
+                    event_subtype=event_subtype,
+                    asset=asset,
+                    balance=Balance(amount),
+                    location_label=self.name,
+                    notes=notes,
+                )
+            elif tx_type in ('incentives_shared_clawback', 'clawback'):
+                return HistoryEvent(
+                    event_identifier=event_identifier,
+                    sequence_index=0,
+                    timestamp=timestamp_ms,
+                    location=Location.COINBASE,
+                    event_type=HistoryEventType.SPEND,
+                    event_subtype=HistoryEventSubType.CLAWBACK,
```

Comment:
for both clawback events is this a spend event from the user?

Reply:
yeah.